central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -554,6 +572,15 @@ func verifyCollectionConstraints(collection *storage.ResourceCollection) error {
 				return errors.Wrapf(errox.InvalidArgs, "unsupported field name %q", selectorRule.GetFieldName())
 			}
 
+			// determine whether this is a label rule
+			var labelRule bool
+			switch selectorRule.GetFieldName() {
+			case pkgSearch.DeploymentLabel.String(), pkgSearch.NamespaceLabel.String(), pkgSearch.ClusterLabel.String():
+				labelRule = true
+			default:
+				labelRule = false
+			}
```

Comment:
seems like good candidate for move this into func `isLabelRule()` and use it here and above.
or change the `supportedFieldNames` to 
```
map[string]struct{
  fieldLabel field.Label
  isLabelSelector bool
}
```